js/foam/dao/GoogleAuthDAO.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.dao',
+  name: 'GoogleAuthDAO',
+  extendsModel: 'foam.dao.ProxyDAO',
+  imports: [
+    'warn'
+  ],
+
+  documentation: function() {/*
+    A DAO decorator that extracts an auth token from the context parameter and
+    checks its validity as a Google OAuth/Google Sign-In token. If it
+    successfully matches, it will set the "sub" claim (an app-specific, stable
+    user ID for this Google account) as the "principal" on the context, for the
+    use of the delegate (probably an AuthenticatedDAO).
+  */},
+
+  constants: {
+    CLOCK_SKEW: 300, // Clock skew tolerance for auth is 5 minutes.
```

Comment:
As discussed offline, cribbed this from Google's own auth module. I've added a link in a comment.